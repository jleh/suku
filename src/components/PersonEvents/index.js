import React from 'react';
import { Translate } from 'react-localize-redux';
import { format } from 'date-fns';

import { renderAge } from '../../util';

import styles from './personEvents.css';

const renderDate = date => date && (date.length === 4 ? date : format(date, 'DD.MM.YYYY'));
const renderAgeOnEvent = (birth, event) => {
  if (!birth) {
    return null;
  }

  return event.type !== 'Birth'
    && renderAge(birth, event.date)
      .replace('vuotta', 'v')
      .replace('päivää', 'pv');
};

export default ({ events, birth }) => {
  if (!events) {
    return null;
  }

  return (
    <table className={styles.table}>
      <tbody>
        {events.map(event => (
          <tr key={event.id}>
            <td className={styles.date}>{renderDate(event.date)}</td>
            <td className={styles.age}>{renderAgeOnEvent(birth, event)}</td>
            <td><Translate id={`events.${event.type}`} /></td>
            <td>{event.description}</td>
            <td className={styles.place}>{event.place && event.place.name}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};
