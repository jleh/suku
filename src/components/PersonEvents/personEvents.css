.timeline {
  margin-top: 1em;
  min-width: 60%;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.timelineRow {
  display: flex;
  flex-direction: row;
  height: 4em;

  & div {
    flex-basis: 6em;
  }
}

.timelineText {
  flex-grow: 2;
  align-self: center;
}

.line {
  max-width: 2em;

  & div {
    height: 100%;
    top: 40%;
    position: relative;
    border-left: 1px solid #000;
  }

  & div::before {
    content: '';
    position: absolute;
    width: 12px;
    height: 12px;
    left: -7px;
    background-color: #fff;
    border-radius: 50%;
    border: 1px solid #000;
  }
}

.place {
  font-style: italic;
  font-size: 0.9em;
}

.age {
  padding-left: 0.5em;
}

.date {
  width: 5em;
  flex-grow: 0.25;
  align-self: center;
}

.last {
  & div {
    border: none;
  }
}
