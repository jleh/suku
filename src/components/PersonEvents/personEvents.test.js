import React from 'react';
import renderer from 'react-test-renderer';

import PersonEvents from '.';

const renderComponent = (events, birth) =>
  renderer.create(<PersonEvents events={events} birth={birth} />).toJSON();

test('Renders nothing if no events', () => {
  const events = undefined;
  const birth = undefined;

  expect(renderComponent(events, birth)).toMatchSnapshot();
});

test('Renders events even if birthday is unknown', () => {
  const events = [
    {
      id: 1,
      date: '1900-06-06',
      type: 'Residence',
      place: { name: 'Place' },
    },
  ];
  const birth = undefined;

  expect(renderComponent(events, birth)).toMatchSnapshot();
});

test('Renders person age on event', () => {
  const events = [
    {
      id: 1,
      date: '1900-06-06',
      type: 'Residence',
      place: { name: 'Place' },
    },
  ];
  const birth = '1850-06-06';

  expect(renderComponent(events, birth)).toMatchSnapshot();
});
