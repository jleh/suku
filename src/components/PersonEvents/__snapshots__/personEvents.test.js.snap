// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders events even if birthday is unknown 1`] = `
<div
  className="timeline"
>
  <div
    className="timelineRow"
  >
    <div
      className="date"
    >
      <div>
        06.06.1900
      </div>
      <div
        className="age"
      />
    </div>
    <div
      className="line last"
    >
      <div />
    </div>
    <div
      className="timelineText"
    >
      <div>
        Missing translationId: events.Residence for language: \${ languageCode }
         
      </div>
      <div
        className="place"
      >
        Place
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders nothing if no events 1`] = `null`;

exports[`Renders person age on event 1`] = `
<div
  className="timeline"
>
  <div
    className="timelineRow"
  >
    <div
      className="date"
    >
      <div>
        06.06.1900
      </div>
      <div
        className="age"
      >
        50 v
      </div>
    </div>
    <div
      className="line last"
    >
      <div />
    </div>
    <div
      className="timelineText"
    >
      <div>
        Missing translationId: events.Residence for language: \${ languageCode }
         
      </div>
      <div
        className="place"
      >
        Place
      </div>
    </div>
  </div>
</div>
`;
